argus/shell.py

```diff
@@ -0,0 +1,261 @@
+                        img_name = image["name"].lower()
+                        break
+                images_list.append((img_id, img_name))
+        finally:
+            mng.cleanup_credentials()
+    elif image_name:
+        try:
+            mng = manager.APIManager()
+            images = mng.image_client.list_images()["images"]
+            for img_name in image_name:
+                img_id = ""
+                for image in images:
+                    if image["name"].lower() == img_name.lower():
+                        img_id = image["id"].lower()
+                        break
+                images_list.append((img_id, img_name))
+        finally:
+            mng.cleanup_credentials()
+
+    # filter unknow images
+    for image in images_list:
+        if not image[0] or not image[1]:
+            images_list.remove(image)
+            print("[ERROR] Invalid image imformation id: {} - name: {}".format(
+                image[0], image[1]))
+
+    return images_list
+
+
+def _get_base_directory(directory):
+    """Return the path to the root directory for this run."""
+    if directory:
+        if not os.path.isdir(directory):
+            os.mkdir(directory)
+        directory = directory
+    else:
+        directory = tempfile.mkdtemp(prefix="argus-", dir="/tmp")
+    return os.path.abspath(directory)
+
+
+def _prepare_argument_parser():
+    """Prepare the argument parser."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-d", "--directory",
+                        help="Directory that will hold the required files.")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-i", "--image_id", action="append",
+                       help="Glance image-id to use for this run.")
+    group.add_argument("-n", "--image_name", action="append",
+                       help="Glance image-name to use for this run."
+                            "This is dangerous, be sure the name matches"
+                            " just one image.")
+    parser.add_argument("-f", "--flavor",
+                        help="The flavor id.")
+    parser.add_argument("-t", "--tests",
+                        help="The tests you want to run.")
+    parser.add_argument("-r", "--resources",
+                        help="URL to argus resources.")
+    parser.add_argument("-p", "--parallel", type=int,
+                        help="many processes to use in parallel.")
+    parser.add_argument("-l", "--local",
+                        help="Local git repository with all the argus files.")
+    parser.add_argument("-s", "--separate", action="store_true",
+                        help="Make sepatate log files for each scenario.")
+    parser.add_argument("-g", "--git_command",
+                        help="The git command that will test the "
+                             "Cloudbase-Init patch.")
+    parser.add_argument("-m", "--msi", help="The msi to use.")
```

Comment:
`default=None` and also please change the name to something more appropriate like `zip_patch` or anything similar, `msi` is misleading regarding the behavior this function has.